compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -2146,15 +2147,19 @@ public void markParameterizedType(BType enclosingType, Collection<BType> constit
 
     // private methods
 
-    private BSymbol resolveOperator(ScopeEntry entry, List<BType> typeList) {
+    private BSymbol resolveOperator(boolean isAdditiveExpr, ScopeEntry entry, List<BType> typeList) {
```

Comment:
This change doesn't seem right to me. This is a genetic method to resolve an operator. If we need to special case something (still not convinced we have to), it has to be done at an upper level.

Reply:
Why can't we fix this in `org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolResolver#getArithmeticOpsForTypeSets`?

Things like

```ballerina
public function main() {
    int:Signed8|int:Signed16 x = 1;
    int _ = x + x;
}
```

already work.